libarchive/write.py

```diff
@@ -73,6 +73,47 @@ def add_files(self, *paths):
                         write_finish_entry(write_p)
                         entry_clear(entry_p)
 
+    def add_entry_from_memory(self, entry_path, entry_size, entry_data):
```

Comment:
@krzysztofzuraw Building your subclass on "private" methods seems like a bad idea to me. It's fragile. Someone (probably me) is going to drop or modify these methods someday and it'll break your code.